### PR TITLE
[DO NOT MERGE] Backport for symfony 2.0 and 2.1

### PR DESCRIPTION
**Pull request by dav-m85**
https://github.com/liip/LiipMonitorBundle/issues/95

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -2,27 +2,16 @@ language: php
...
-    - php: 5.6
-      env: SYMFONY_VERSION=2.4.*
-    - php: 5.6
-      env: SYMFONY_VERSION=2.5.*
-    - php: 5.6
-      env: SYMFONY_VERSION=2.7.*@dev
+    - php: 5.3
+      env: SYMFONY_VERSION=2.0.*
+    - php: 5.3
+      env: SYMFONY_VERSION=2.1.*
```
Comment (dav-m85) [inline .travis.yml]: Obvious, but I think it could work with newer version as well.

File: Command/HealthCheckCommand.php [hunk 2]
```diff
@@ -28,6 +29,12 @@ protected function configure()
                     InputOption::VALUE_REQUIRED | InputOption::VALUE_IS_ARRAY,
                     'Additional reporters to run.',
                     array()
+                ),
+                new InputOption(
```
Comment (dav-m85) [inline Command/HealthCheckCommand.php]: that's a new feature, it makes sure the cli command outputs error codes suitable for nagios (exitCode 0 : ok, 1: got warnings, 2: got criticals)

File: Command/HealthCheckCommand.php [hunk 3]
```diff
@@ -36,13 +43,34 @@ protected function execute(InputInterface $input, OutputInterface $output)
     {
         $checkName = $input->getArgument('checkName');
         $runner = $this->getContainer()->get('liip_monitor.runner');
-        $runner->addReporter(new ConsoleReporter($output));
+
+        if ($input->getOption('nagios')) {
+            $runner->addReporter(new RawConsoleReporter($output));
```
Comment (dav-m85) [inline Command/HealthCheckCommand.php]: Raw console reporter just outputs like console reporter but without the console coloring.

File: DependencyInjection/Configuration.php [hunk 4]
```diff
@@ -30,7 +30,6 @@ public function getConfigTreeBuilder()
                 ->booleanNode('enable_controller')->defaultFalse()->end()
                 ->scalarNode('view_template')->defaultNull()->end()
                 ->arrayNode('mailer')
-                    ->canBeEnabled()
```
Comment (dav-m85) [inline DependencyInjection/Configuration.php]: Mainly removed in this file all sf2.2 new configurations additions.

File: README.md [hunk 5]
```diff
@@ -30,9 +36,9 @@ delivers the results in JSON format.
 
 Here's the web interface:
 
-![Web Interface](http://img.skitch.com/20120312-fhyc74ese9jjpyd1wxjcdbs85b.jpg "Web Interface")
+@todo Interface screenshot to provide.
```
Comment (dav-m85) [inline README.md]: Not true anymore.

File: README.md [hunk 6]
```diff
@@ -30,9 +36,9 @@ delivers the results in JSON format.
...
 
-## Installation ##
+## Installation
```
Comment (dav-m85) [inline README.md]: Personal taste, not sure you'll like it.

File: Command/HealthCheckCommand.php (hunk 3 above)
Comment (lsmith77) [inline Command/HealthCheckCommand.php]: this is an addition that could/should also go into master?

**Comment by lsmith77** [comment]
if I see this properly .. the key change is with the Configuration class. the rest of the changes could also be interesting to master. now one approach to this could be maintaining a second configuration class that we use in case Symfony 2.0/2.1 is used. would be a bit ugly to have this in master and like I said, 2.0/2.1 is very dead for most people.

/cc @kbond

**Comment by dav-m85** [comment]
It would be a pleasure to split the PR into feature pieces. I don't know how to proceed to maintain two configurations, any insights ? That was why I first went for the different branches.

**Comment by dav-m85** [comment]
btw, we still have 2.0 servers serving 10M requests per month, not so dead ;)

**Comment by kbond** [comment]
Interesting.. but yeah.. not quite sure where to put it. A 1.1 branch could cause bc breaks for people using `~1.0`. I don't particularly like having a separate duplicated `Configuration` in master but I don't know how else to handle this..

Let's definitely get PRs for the other changes though.

**Comment by lsmith77** [comment]
maybe in the end the best approach is that we bring in all your general improvements and you just keep a fork with a modified composer.json and Configuration class which users can pull in using an alias in their composer.json. I am open to mentioning your fork in the official docs. This way it should be a minimal effort on your part to maintain your fork and we can easily stay functionally in sync.

**Comment by dav-m85** [comment]
This sounds perfect. I'll push the new features to review today !

**Comment by dav-m85** [comment]
I extracted the nagios support feature here: https://github.com/liip/LiipMonitorBundle/pull/97

**Pull request closed without merging**